clang: error: unsupported option '-fopenmp'

👋 trying to build the latest release, but run into some build issue. The error log is as below:


<details>
<summary>error build log</summary>

```
gmake[2]: *** [modules/graph/CMakeFiles/vineyard_graph.dir/build.make:149: modules/graph/CMakeFiles/vineyard_graph.dir/fragment/arrow_fragment_builder_string_uint64.cc.o] Error 1
clang: error: unsupported option '-fopenmp'
clang: error: unsupported option '-fopenmp'
gmake[2]: *** Waiting for unfinished jobs....
clang: error: unsupported option '-fopenmp'
clang: error: unsupported option '-fopenmp'
gmake[2]: *** [modules/graph/CMakeFiles/vineyard_graph.dir/build.make:205: modules/graph/CMakeFiles/vineyard_graph.dir/fragment/arrow_fragment_string.cc.o] Error 1
clang: error: unsupported option '-fopenmp'
```

</details>

relates to Homebrew/homebrew-core#157199

--- a/modules/graph/CMakeLists.txt
+++ b/modules/graph/CMakeLists.txt
@@ -63,8 +63,10 @@ file(GLOB_RECURSE GRAPH_SRC_FILES "${CMAKE_CURRENT_SOURCE_DIR}" "fragment/*.cc"
 
 add_library(vineyard_graph ${GRAPH_SRC_FILES} ${powturbo-target-objects})
 target_add_debuginfo(vineyard_graph)
-target_compile_options(vineyard_graph PUBLIC "-fopenmp")
-target_link_options(vineyard_graph PUBLIC "-fopenmp")
+if(NOT APPLE)
+    target_compile_options(vineyard_graph PUBLIC "-fopenmp")
+    target_link_options(vineyard_graph PUBLIC "-fopenmp")
+endif()
 target_include_directories(vineyard_graph PUBLIC ${MPI_CXX_INCLUDE_PATH})
 
 find_package(Boost COMPONENTS leaf)

